api/circulation.py

```diff
@@ -342,6 +342,15 @@ def borrow(self, patron, pin, licensepool, delivery_mechanism,
             )
 
         new_loan = False
+
+        # TODO: If the book isn't available and the patron is putting it on hold,
+        # we shouldn't check the loan limit. Or should there be one limit for loans
+        # and holds?
```

Comment:
Separate loan and hold limits is the way to go, I think. I agree this is a TODO but it's kind of a marginal case where you're at your loan limit but not at your hold limit, and you try to borrow a book but end up putting it on hold.

Reply:
I agree. But this code also runs when you know a book's not available and try to put it on hold - we try a loan first even if we expect it won't work.

Reply:
You're right, that's a different case. If you try to put a book on hold and you're over your loan limit, it should try to put the book on hold, without trying to borrow it first. If the hold fails because the book is actually available, that's the time to raise PatronLoanLimitReached.